### PR TITLE
fix: use string not object in LemonSelect

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We are using objects not string as values in the `LemonSelect` component. Causing linting errors in https://github.com/PostHog/posthog/pull/17755

## Changes

Refactor some data warehouse code to support the table name rather than the entire object

## How did you test this code?

Manually & with Typescript

### CODE REVIEW

**Comment by daibhin** [comment]
@EDsCODE I'm working on upgrading React and part of that means stricter typing. This produced typing errors in https://github.com/PostHog/posthog/pull/17731 because an object was being passed in as a `value` in the LemonSelect. We typically use primitive values for [comparison purposes](https://github.com/PostHog/posthog/blob/master/frontend/src/lib/lemon-ui/LemonSelect/LemonSelect.tsx#L238) and for [rendering](https://github.com/PostHog/posthog/blob/master/frontend/src/lib/lemon-ui/LemonSelect/LemonSelect.tsx#L149).

I went ahead and migrated it to work exactly as before but wanted to get your 👀 on it

File: frontend/src/scenes/data-warehouse/ViewLinkModal.tsx [hunk 1]
```diff
@@ -35,8 +35,9 @@ interface ViewLinkFormProps {
 }
 
 export function ViewLinkForm({ tableSelectable }: ViewLinkFormProps): JSX.Element {
-    const { viewOptions, toJoinKeyOptions, selectedView, selectedTable, fromJoinKeyOptions } = useValues(viewLinkLogic)
-    const { selectView, toggleFieldModal, selectTable } = useActions(viewLinkLogic)
+    const { viewOptions, toJoinKeyOptions, selectedView, selectedTableName, fromJoinKeyOptions } =
+        useValues(viewLinkLogic)
+    const { selectView, toggleFieldModal, selectTableName } = useActions(viewLinkLogic)
     const { tableOptions } = useValues(databaseSceneLogic)
```
Comment (EDsCODE) [inline frontend/src/scenes/data-warehouse/ViewLinkModal.tsx:41]: NIT: can import this from viewLinkLogic now!

**Review by EDsCODE — APPROVED** [review]
lgtm!

File: frontend/src/scenes/data-warehouse/ViewLinkModal.tsx (hunk 1 above)
Comment (daibhin) [reply frontend/src/scenes/data-warehouse/ViewLinkModal.tsx:41]: Perfect, updated!

**Pull request merged**